aiohttp/connector.py

```diff
@@ -1063,7 +1063,13 @@ async def _create_proxy_connection(
                 proxy_req.headers[hdrs.PROXY_AUTHORIZATION] = auth
 
         if req.is_ssl():
-            sslcontext = self._get_ssl_context(req)
+            # Safety of the `cast()` call here is based on the fact that
+            # internally `_get_ssl_context()` only returns `None` when
+            # `req.is_ssl()` evaluates to `False` which is never gonna happen
+            # in this code path. Of course, it's rather fragile
+            # maintainability-wise but this is to be solved separately.
+            sslcontext = cast(ssl.SSLContext, self._get_ssl_context(req))
```

Comment:
(mypy knows that this may return `None` but we know that this won't happen under `if req.is_ssl():`)